chore(test-autofix): Prevent source pollution in test

nbrahms opened the pull request:
Previously, tests would intermittently fail due to pollution of
`cli/tests/e2e/targets`. This was caused by `test_autofix` creating
a temporary file within this path at test run time.

Issue was reproduced using `chmod -R 0400 cli/tests/e2e/targets`, then
running `pytest`.

This commit updates test_autofix to run entirely in a copied file-system
context, thereby avoiding any pollution of the test sources. As a bonus,
the test_autofix code has been drastically simplified and runs 2x
faster.

I've also renamed `run_semgrep_no_symlink` to
`run_semgrep_on_copied_files` for developer clarity.

This commit also updates snapshots to reflect the actual output of
semgrep when `--dryrun` is ommitted from its invocation. (Note that the
output with `--dryrun` is still captured in a corresponding set of
snapshots).

No changelog needed with this commit.

PR checklist:

- [x] Change has no security implications (otherwise, ping security team)

Commit:
chore(test-autofix): Prevent source pollution in test

Previously, tests would intermittently fail due to pollution of
`cli/tests/e2e/targets`. This was caused by `test_autofix` creating
a temporary file within this path at test run time.

Issue was reproduced using `chmod -R 0400 cli/tests/e2e/targets`, then
running `pytest`.

This commit updates test_autofix to run entirely in a copied file-system
context, thereby avoiding any pollution of the test sources. As a bonus,
the test_autofix code has been drastically simplified and runs 2x
faster.

I've also renamed `run_semgrep_no_symlink` to
`run_semgrep_on_copied_files` for developer clarity.

No changelog needed with this commit.

mmcqd commented on cli/tests/e2e/snapshots/test_autofix/test_autofix/rulesautofixtwo-autofixes.yaml-autofixtwo-autofixes.txt-not-dryrun/results.json:
```diff
@@ -44,9 +41,6 @@
       "extra": {
         "fingerprint": "36e519fa759ca74dea55c0aef65e0e3e081cdfeb4370a8b8a2546e77f65952ead0e128ca7b79e10645edaac3006b670ff0e122e96ee1569dca4105167a8b6b7e_0",
         "fix": "four",
-        "fixed_lines": [
```
We've lost all the `fixed_lines` in autofix output. Semgrep app uses those.

nbrahms replied:
@mmcqd we actually haven't, they were just duplicated. See the squash commit message:

> This commit also updates snapshots to reflect the actual output of
semgrep when `--dryrun` is ommitted from its invocation. (Note that the
output with `--dryrun` is still captured in a corresponding set of
snapshots).

mmcqd replied:
ack, reading is hard

Commit:
squash! chore(test-autofix): Prevent source pollution in test

This commit also updates snapshots to reflect the actual output of
semgrep when `--dryrun` is ommitted from its invocation. (Note that the
output with `--dryrun` is still captured in a corresponding set of
snapshots).
